src/fastgltf_tools.hpp

```diff
@@ -0,0 +1,467 @@
+#if FASTGLTF_HAS_CONCEPTS
+requires Element<ElementType>
+#endif
+ElementType getAccessorElementAt(ComponentType componentType, const std::byte* bytes) {
+	switch (componentType) {
+		// This is undefined behavior if component type is invalid
+		default:
+		case ComponentType::Byte:
+			return convertAccessorElement<ElementType, std::int8_t>(bytes, Seq{});
+		case ComponentType::UnsignedByte:
+			return convertAccessorElement<ElementType, std::uint8_t>(bytes, Seq{});
+		case ComponentType::Short:
+			return convertAccessorElement<ElementType, std::int16_t>(bytes, Seq{});
+		case ComponentType::UnsignedShort:
+			return convertAccessorElement<ElementType, std::uint16_t>(bytes, Seq{});
+		case ComponentType::Int:
+			return convertAccessorElement<ElementType, std::int32_t>(bytes, Seq{});
+		case ComponentType::UnsignedInt:
+			return convertAccessorElement<ElementType, std::uint32_t>(bytes, Seq{});
+		case ComponentType::Float:
+			return convertAccessorElement<ElementType, float>(bytes, Seq{});
+		case ComponentType::Double:
+			return convertAccessorElement<ElementType, double>(bytes, Seq{});
+	}
+}
+
+// Performs a binary search for the index into the sparse index list whose value matches the desired index
+template <typename ElementType>
+bool findSparseIndex(const std::byte* bytes, std::size_t indexCount, std::size_t desiredIndex,
+		std::size_t& resultIndex) {
+	auto* elements = reinterpret_cast<const ElementType*>(bytes);
+	auto* last = &reinterpret_cast<const ElementType*>(bytes)[indexCount];
+	auto count = static_cast<std::ptrdiff_t>(indexCount);
+
+	resultIndex = 0;
+
+	while (count > 0) {
+		auto step = count / 2;
+		auto index = resultIndex + step;
+
+		if (elements[index] < static_cast<ElementType>(desiredIndex)) {
+			resultIndex = index + 1;
+			count -= step + 1;
+		} else {
+			count = step;
+		}
+	}
+
+	return resultIndex < indexCount && elements[resultIndex] == static_cast<ElementType>(desiredIndex);
+}
+
+// Finds the index of the nearest sparse index to the desired index
+bool findSparseIndex(ComponentType componentType, const std::byte* bytes, std::size_t indexCount,
+		std::size_t desiredIndex, std::size_t& resultIndex) {
+	switch (componentType) {
+		case ComponentType::Byte:
+			return findSparseIndex<std::int8_t>(bytes, indexCount, desiredIndex, resultIndex);
+		case ComponentType::UnsignedByte:
+			return findSparseIndex<std::uint8_t>(bytes, indexCount, desiredIndex, resultIndex);
+		case ComponentType::Short:
+			return findSparseIndex<std::int16_t>(bytes, indexCount, desiredIndex, resultIndex);
+		case ComponentType::UnsignedShort:
+			return findSparseIndex<std::uint16_t>(bytes, indexCount, desiredIndex, resultIndex);
+		case ComponentType::Int:
+			return findSparseIndex<std::int32_t>(bytes, indexCount, desiredIndex, resultIndex);
+		case ComponentType::UnsignedInt:
+			return findSparseIndex<std::uint32_t>(bytes, indexCount, desiredIndex, resultIndex);
+		default:
+			break;
+	}
+
+	return false;
+}
+
+}
```

Comment:
`} // namespace internal`

Reply:
Fixed in [df17edc](https://github.com/spnda/fastgltf/pull/20/commits/df17edc742f1893ece80c0cc2467a9b68b7ec880)